### ISSUE TITLE
Separate matching and filtering logic in background_peaks_update

### ISSUE THREAD

**Issue by PMSeitzer** [eugenemel/maven#25]
`background_peaks_update.cpp` has a method, `matchFragmentation(Peakgroup*)`, which matches the peakgroup a database and performs some filtering of results.

some filtering of results is performed downstream.

Instead, the method should simply return matches, and all filtering should be handled downstream.

**Comment by PMSeitzer** [comment]
This has been handled as part of resolving Issue #24 (on branch `issue-24`).

**Comment by PMSeitzer** [comment]
This issue should be closed once Issue #24 is closed (and branch `issue-24` is merged into `master` and deleted.
